src/hyperlight_host/src/hypervisor/hyperv_linux.rs

```diff
@@ -89,7 +105,23 @@ impl HypervLinuxDriver {
         }
         let mshv = Mshv::new()?;
         let pr = Default::default();
+        #[cfg(mshv2)]
         let vm_fd = mshv.create_vm_with_config(&pr)?;
+        #[cfg(mshv3)]
+        let vm_fd = {
+            // It's important to avoid create_vm() and explicitly use
+            // create_vm_with_args() with an empty arguments structure
+            // here, because otherwise the partition is set up with a SynIC.
+
+            let vm_fd = mshv.create_vm_with_args(&pr)?;
+            let features: hv_partition_synthetic_processor_features = Default::default();
+            vm_fd.hvcall_set_partition_property(
+                hv_partition_property_code_HV_PARTITION_PROPERTY_SYNTHETIC_PROC_FEATURES,
+                unsafe { features.as_uint64[0] },
+            )?;
+            vm_fd
+        };
+
```

Comment:
we need a vm_fd.initialize()?; call (at least for mshv3, i didn't test this on mshv2)
thanks @syntactically for pointing this out